### PR TITLE
Reduce memory usage, allow selection of multiple metrics in MetricsPane [HZG-208]

### PR DESCRIPTION
**Pull request by hazelcast-alexb**
This PR reduces the memory footprint while loading large diagnostics files.

Previously each individual `DiagnosticFile` contained an index for the metric entries contained within it. Now there is a single index per instance of `InstanceDiagnostics`, this hugely reduces the number of index entries needed and allows the tool to load several gigabytes of diagnostic logs.

In addition a number of IDE compiler warnings have been fixed and the `MetricsPane` has been updated to allow selection and comparison of multiple metrics.

### CODE REVIEW

File: src/main/java/com/hazelcast/diagnostics/InvocationProfilerPane.java [hunk 1]
```diff
@@ -12,16 +12,21 @@
...
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Set;
 import java.util.SortedMap;
 import java.util.TreeMap;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
```
Comment (JackPGreen) [inline src/main/java/com/hazelcast/diagnostics/InvocationProfilerPane.java:29]:
```suggestion
import java.util.ArrayList;
import java.util.Collection;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;
import java.util.Map.Entry;
import java.util.SortedMap;
import java.util.TreeMap;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
```

I don't think any of these are needed.

File: src/main/java/com/hazelcast/diagnostics/InstanceDiagnostics.java [hunk 2]
```diff
@@ -13,33 +13,46 @@
...
+
+    public static final String METRIC_OPERATION_INVOCATIONS_PENDING = "[unit=count,metric=operation.invocations.pending]";
+    public static final String METRIC_OPERATION_QUEUE_SIZE = "[unit=count,metric=operation.queueSize]";
+    public static final String METRIC_RUNTIME_USED_MEMORY = "[metric=runtime.usedMemory]";
+    public static final String METRIC_OPERATION_INVOCATIONS_LAST_CALL_ID = "[unit=count,metric=operation.invocations.lastCallId]";
+    public static final String METRIC_OPERATION_COMPLETED_COUNT = "[unit=count,metric=operation.completedCount]";
+    public static final String METRIC_MEMORY_USED_HEAP = "[unit=bytes,metric=memory.usedHeap]";
+    public static final String METRIC_MEMORY_MAX_HEAP = "[unit=bytes,metric=memory.maxHeap]";
+    public static final String METRIC_OS_PROCESS_CPU_LOAD = "[metric=os.processCpuLoad]";
+    public static final String METRIC_RUNTIME_TOTAL_MEMORY = "[metric=runtime.totalMemory]";
```
Comment (JackPGreen) [inline src/main/java/com/hazelcast/diagnostics/InstanceDiagnostics.java:47]: Could these be an `enum`?

**Review by JackPGreen — CHANGES_REQUESTED** [review]

File: src/main/java/com/hazelcast/diagnostics/InstanceDiagnostics.java (hunk 2 above)
Comment (JamesHazelcast) [reply src/main/java/com/hazelcast/diagnostics/InstanceDiagnostics.java:47]: I think this is fine as-is, @hazelcast-alexb has mainly compiled strings used all over the place in this project into constants, and then added to it - I don't think there's much value to be gained from making these enums

**Review by JamesHazelcast — APPROVED** [review]
LGTM, we should probably clean up the unused imports before merging, but approving in advance. Thanks for this @hazelcast-alexb!

**Review by JackPGreen — APPROVED** [review]
Some silly comments from me, but it's totally fine as it is :)

File: src/main/java/com/hazelcast/diagnostics/InstanceDiagnostics.java [hunk 3]
```diff
@@ -220,163 +185,167 @@ public String fixMetricName(String metricName) {
...
+                }
+            }
         }
 
-        if(metricName.equals("[unit=count,metric=operation.invocations.lastCallId]")){
-            return "operation.invocations.lastCallId";
+        public static DiagnosticType from(String prefix) {
+            DiagnosticType type = byPrefix.get(prefix);
+            if( type == null)
+            {
```
[the remark this answers is not in the document]
Comment (hazelcast-alexb) [reply src/main/java/com/hazelcast/diagnostics/InstanceDiagnostics.java:245]: Lol. That's a hangover from the last place's code style! Old habits die hard.

**Pull request merged**